### DISCUSSION TITLE
How to handle many web views that only differ slightly?

### DISCUSSION THREAD

**Discussion by XaverStiensmeier** [TriliumNext/Trilium#4628] [Q&A] [answered]
There's an online database where you can get data by using link.link/{id}. I would like to be able to include web views to specific items. Do I have to create a web view for every single item or is it possible to create the web view on the fly once a link in a note is clicked?

I am aware that I can probably use the backend API to create the view, but I am unsure whether that's the best way.

**Comment by meichthys** [comment]
I'm not sure what you mean by web-view, but you may be able to leverage https://github.com/rauenzi/Trilium-SingleFile , but you will likely need to run each URL separately. This is currently out of scope to be included in Trilium natively.

**Reply by XaverStiensmeier** [reply]
I am referring with "web-view" to the node type "Web View" that allows you to include a browser like view of a page. I feel like SingleFile is not able to offer anything beyond "Web View" when it comes to online pages. I might be mistaken.
![image](https://github.com/zadam/trilium/assets/36056823/2da58a73-f578-481b-8877-c3b52060d79c)

**Reply by meichthys** [reply, marked as answer]
I see what you mean now. There is no automated way to import multiple links at once, and likely will not be a way in the near future.

**Reply by XaverStiensmeier** [reply]
Thanks. I have opened a discussion on handling get-requests in trilium https://github.com/zadam/trilium/discussions/4630 which will lead me to implementing a method for it - at least for my specific problem - that might be generalized by others.
